swagger.yaml

```diff
@@ -1,75 +1,298 @@
+            - "skill"
+            - "power"
+            - "status"
+            - "curse"
+            default: "attack,skill,power,status,curse"
+          collectionFormat: "multi"
+
+        - name: "name"
+          in: "query"
+          description: "Card name"
+          required: false
+          type: "string"
+
+        - name: "color"
+          in: "query"
+          description: "List of colors of cards"
+          required: false
+          type: "array"
+          items:
+            type: "string"
+            enum:
+            - "red"
+            - "green"
+            - "blue"
+            - "colorless"
+            default: "red,green,blue,colorless"
+          collectionFormat: "multi"
+
+        - name: "rarity"
+          in: "query"
+          description: "List of rarities of cards"
+          required: false
+          type: "array"
+          items:
+            type: "string"
+            enum:
+            - "basic"
+            - "common"
+            - "uncommon"
+            - "rare"
+            default: "basic,common,uncommon,rare"
+          collectionFormat: "multi"
+
+        - name: "energyMin"
+          in: "query"
+          description: "Minimum value of energy costs of cards (X = 0). Used with energyMax"
+          required: false
+          type: "integer"
+
+        - name: "energyMax"
+          in: "query"
+          description: "Maximum value of energy costs of cards (X = 0). Used with energyMin"
+          required: false
+          type: "integer"
+
+        - name: "keywords"
+          in: "query"
+          description: "List of keywords found in a card's description"
+          required: false
+          type: "array"
+          items:
+            type: "string"
+
+        - name: "number"
+          in: "query"
+          description: "Number of cards to return"
+          required: false
+          type: integer
+          default: 10
+
       responses:
         200:
-          description: Object containing information
+          description: "Cards successfully retrieved"
+          schema:
+            $ref: "#/definitions/MultiData"
+        400:
+          description: "Invalid request"
+          schema:
+            $ref: "#/definitions/Error"
+        404:
+          description: "Card not found"
+          schema:
+            $ref: "#/definitions/Error"
+        500:
+          description: "Internal server error"
+          schema:
+            $ref: "#/definitions/Error"
+
+    post:
+      tags:
+      - "cards"
+      summary: "Add a new card to the database"
+      description: ""
+      operationId: "addCard"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+        - name: "body"
+          in: "body"
+          description: "Card object to be added to the database"
+          required: true
+          schema:
+            $ref: "#/definitions/SingleData"
+
+      responses:
+        201:
```

Comment:
For POST and PUT, it would be a good idea to return the object that was added or updated in the http response.